src/dune/simple_rules.ml

```diff
@@ -169,16 +169,24 @@ let copy_files sctx ~dir ~expander ~src_dir (def : Copy_files.t) =
     Build_system.eval_pred
       (File_selector.create ~dir:(Path.build src_in_build) pred)
   in
-  Path.Set.map files ~f:(fun file_src ->
-      let basename = Path.basename file_src in
-      let file_dst = Path.Build.relative dir basename in
-      SC.add_rule sctx ~loc ~dir
-        (( if def.add_line_directive then
-           Build.copy_and_add_line_directive
-         else
-           Build.copy )
-           ~src:file_src ~dst:file_dst);
-      Path.build file_dst)
+  let targets =
+    Path.Set.map files ~f:(fun file_src ->
+        let basename = Path.basename file_src in
+        let file_dst = Path.Build.relative dir basename in
+        SC.add_rule sctx ~loc ~dir ~mode:def.mode
+          (( if def.add_line_directive then
+             Build.copy_and_add_line_directive
+           else
+             Build.copy )
+             ~src:file_src ~dst:file_dst);
+        Path.build file_dst)
+  in
+  Option.iter
+    ~f:(fun alias ->
+      let alias = Alias.make alias ~dir in
+      Rules.Produce.Alias.add_deps alias targets)
+    def.alias;
```

Comment:
Could you move def.alias to be the first argumnt? Makes it simpler to read.

Reply:
Thanks, it is fixed.